modules/openapi-generator/src/main/resources/dart/libraries/dio/serialization/built_value/class_serializer.mustache

```diff
@@ -10,26 +10,17 @@ class _${{classname}}Serializer implements PrimitiveSerializer<{{classname}}> {
     {{{classname}}} object, {
     FullType specifiedType = FullType.unspecified,
   }) sync* {
-    {{#vendorExtensions.x-self-and-ancestor-only-props}}
-    {{#required}}
-    {{!
-        A required property need to always be part of the serialized output.
-        When it is nullable, null is serialized, otherwise it is an error if it is null.
-    }}
+    {{#vendorExtensions.x-self-and-ancestor-only-props}}    
+    {{! Non-required properties are only serialized if not null. }}
+    {{^required}}
+    if (object.{{{name}}} != null) {
+    {{/required}}
     yield r'{{baseName}}';
-    yield {{#isNullable}}object.{{{name}}} == null ? null : {{/isNullable}}serializers.serialize(
+    yield {{#required}}{{#isNullable}}object.{{{name}}} == null ? null : {{/isNullable}}{{/required}}serializers.serialize(
```

Comment:
Wonder if it is more readable to split this all into a `required` and  `^required` block which would also avoid all the formatting changes?

Reply:
I think it's better if we don't repeat the logic, and minimize the blocks to the actual changes

Reply:
Yea I would usually agree but all these formatting changes to unrelated files make it hard to really review this.

Reply:
ok I will revert this change for now, since it's actually unrelated to the PR